clang/include/clang/Basic/Attr.td

```diff
@@ -1415,13 +1415,33 @@ def LoopUnrollHint : StmtAttr {
 }
 
 def IntelReqdSubGroupSize: InheritableAttr {
-  let Spellings = [GNU<"intel_reqd_sub_group_size">,
-                   CXX11<"intel", "reqd_sub_group_size">];
+  let Spellings = [
+    GNU<"intel_reqd_sub_group_size">, CXX11<"intel", "reqd_sub_group_size">,
+    CXX11<"intel", "sub_group_size"> // SYCL2020 spelling.
+  ];
   let Args = [ExprArgument<"Value">];
   let Subjects = SubjectList<[Function], ErrorDiag>;
   let Documentation = [IntelReqdSubGroupSizeDocs];
   let LangOpts = [OpenCL, SYCLIsDevice, SilentlyIgnoreSYCLIsHost];
   let SupportsNonconformingLambdaSyntax = 1;
+  let AdditionalMembers = [{
+    bool isSYCL2020Spelling() const {
+      return getSemanticSpelling() == CXX11_intel_sub_group_size;
+    }
+  }];
+}
+
+def IntelNamedSubGroupSize : InheritableAttr {
+  let Spellings = [CXX11<"intel", "named_sub_group_size">];
+  let Args = [IdentifierArgument<"Type">];
```

Comment:
This seems like it should be an `EnumArgument` given that arbitrary identifiers aren't allowed.

Reply:
Ooh yeah!  Good idea.  I still probably have to add the 'kw as identifier' bits, but enum makes more sense, and creates a nicer customization point.